### PR TITLE
 BaSP-MR-27:CLASS TEST- get getById and create

### PR DESCRIPTION
**Pull request by Franro212**
Unit Test
Create test for get method.
Create test for post method.
Create test for getById method.

### CODE REVIEW

File: src/tests/classes.test.js [hunk 1]
```diff
@@ -1,5 +1,95 @@
...
+afterEach(async () => {
+  await Class.collection.deleteMany();
+});
+
+describe('GET /api/classes', () => {
+  test('should return status 200 and valid JSON response', async () => {
+    const response = await request(app).get('/api/classes');
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.message).toBe('Classes found successfully!');
```
Comment (JuanManuel-Blois) [inline src/tests/classes.test.js:37]: we should have an expect to check the data of the body

File: src/tests/classes.test.js [hunk 2]
```diff
@@ -1,5 +1,95 @@
...
+  });
+
+  test('should return status 400 for invalid route', async () => {
+    const response = await request(app).get('/api/classes/invalid');
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.message).toBe('This is not a valid object Id');
+  });
```
Comment (JuanManuel-Blois) [inline src/tests/classes.test.js:45]: we should have an expect to check the data of the body

File: src/tests/classes.test.js [hunk 3]
```diff
@@ -1,5 +1,95 @@
...
+});
+
+describe('POST /api/classes/', () => {
+  test('should create a new class and return a status 201', async () => {
+    const response = await request(app).post('/api/classes').send(mockClass);
+    expect(response.status).toBe(201);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.message).toBe('Class created successfully!');
```
Comment (JuanManuel-Blois) [inline src/tests/classes.test.js:53]: we should have an expect to check the data of the body

File: src/tests/classes.test.js [hunk 4]
```diff
@@ -1,5 +1,95 @@
...
+    expect(response.body.message).toBe('Classes found successfully!');
+  });
+
+  test('should return status 400 for invalid route', async () => {
+    const response = await request(app).get('/api/classes/invalid');
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
+    expect(response.body.message).toBe('This is not a valid object Id');
+  });
+});
```
Comment (JuanManuel-Blois) [inline src/tests/classes.test.js:46]: we should also have a test where we dont have any classes created, try it with the deleteMany, we also need a test for the error 500

File: src/tests/classes.test.js [hunk 5]
```diff
@@ -1,5 +1,95 @@
...
+    const response = await request(app).post('/api/classes').send(mockClass);
+    expect(response.status).toBe(201);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.message).toBe('Class created successfully!');
+  });
+
+  test('should return status 400 for invalid request data', async () => {
+    const response = await request(app).post('/api/classes').send(mockInvalidClass);
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBeTruthy();
```
Comment (JuanManuel-Blois) [inline src/tests/classes.test.js:59]: we should add expects for the message and data of the body

File: src/tests/classes.test.js [hunk 6]
```diff
@@ -1,5 +1,95 @@
...
+});
+
+describe('POST /api/classes/', () => {
+  test('should create a new class and return a status 201', async () => {
+    const response = await request(app).post('/api/classes').send(mockClass);
+    expect(response.status).toBe(201);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body.message).toBe('Class created successfully!');
+  });
+
```
Comment (JuanManuel-Blois) [inline src/tests/classes.test.js:55]: we should add another test where we check if a field is missing in the seed, you have already created a mock variable for this, this one mockInvalidClass

**Review by LucianoAlarcon — APPROVED** [review]

File: src/tests/classes.test.js (hunk 1 above)
Comment (Franro212) [reply src/tests/classes.test.js:38]: done

File: src/tests/classes.test.js (hunk 2 above)
Comment (Franro212) [reply src/tests/classes.test.js:47]: done

File: src/tests/classes.test.js (hunk 3 above)
Comment (Franro212) [reply src/tests/classes.test.js:73]: done

File: src/tests/classes.test.js (hunk 4 above)
Comment (Franro212) [reply src/tests/classes.test.js:66]: done

File: src/tests/classes.test.js (hunk 5 above)
Comment (Franro212) [reply src/tests/classes.test.js:86]: done

File: src/tests/classes.test.js (hunk 6 above)
Comment (Franro212) [reply src/tests/classes.test.js:82]: done

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**